### PR TITLE
Address @casz review comments from EWM PR#68 (JCasC support)

### PR DESCRIPTION
**Pull request by BenjaminBeggs**
This PR will attempt to address all feedback from @casz for https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68. Once the work is finished and the changes approved, this PR can be merged by a user with write access on this repository to have the commits contribute directly to https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68.

**Review comments addressed**:
- Deprecated ConfigAsCodeTest class replaced with ConfiguredWithCode annotation to load .yaml configuration.
- JenkinsConfiguredWithCodeRule replaces JenkinsRule.
- Deleted unused "obsolete-configuration-as-code.yaml" file
- Export configuration unit tests have been rewritten
- Determined cause of commented out node configuration error (https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68#discussion_r303040400 caused by indentation inconsistencies and `jnlp:` line)

@martinda @imaffe

### CODE REVIEW

**Review by martinda — APPROVED** [review]
Looks good to me.

**Comment by BenjaminBeggs** [comment]
If you feel comfortable with the changes you can merge them to have the commits contribute to https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68 (JCasC support for EWM).

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
 
     @ClassRule public static JenkinsRule r = new JenkinsRule();
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
    @Test
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 2]
```diff
@@ -59,7 +60,7 @@ public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception
     }
 
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
    @Test
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 3]
```diff
@@ -83,7 +84,7 @@ public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
         assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template21"));
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
    @Test
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 4]
```diff
@@ -97,7 +98,7 @@ public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
         assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getNodeMountPoint(), is("/tmp/master-node"));
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
    @Test
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 5]
```diff
@@ -108,7 +109,7 @@ public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
         List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
    @Test
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 6]
```diff
@@ -97,7 +98,7 @@ public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
         assertThat(nodeDiskPools.get(0).getNodeDisks().get(0).getNodeMountPoint(), is("/tmp/master-node"));
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]: 102-104 can be removed 😓 
```suggestion
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 7]
```diff
@@ -83,7 +84,7 @@ public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
         assertThat(nodeDisk.getNodeMountPoint(), is("/tmp/template21"));
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]: 88-90 can be removed 😓 
```suggestion
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 8]
```diff
@@ -59,7 +60,7 @@ public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception
     }
 
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]: 64-66 can be removed 😓 
```suggestion
```

**Comment by jetersen** [comment]
This all wrong 😆 You basically removed the tests 😓

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 9]
```diff
@@ -40,7 +41,7 @@
 
     @ClassRule public static JenkinsRule r = new JenkinsRule();
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]:
```suggestion
     @ClassRule
     @ConfiguredWithCode("configuration-as-code.yml")
     public static JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
```

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java [hunk 10]
```diff
@@ -108,7 +109,7 @@ public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
         List<NodeDiskPool> nodeDiskPools = node.getNodeProperties().get(ExternalWorkspaceProperty.class).getNodeDiskPools();
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void exportConfiguration() throws Exception {
```
Comment (jetersen) [inline src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]: This test needs to be completely rewritten

**Review by jetersen — CHANGES_REQUESTED** [review]
See this for reference on an import/export test with a `@ClassRule`
https://github.com/jenkinsci/slack-plugin/blob/8acdef301a75b348f4d20794a4fd31c439c669b6/src/test/java/jenkins/plugins/slack/jcasc/ConfigurationAsCodeTest.java

**Comment by BenjaminBeggs** [comment]
Haha! Thanks for the review, I’ll sort this out.

File: src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java (hunk 10 above)
Comment (BenjaminBeggs) [reply src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java]: I will rewrite this.

**Comment by BenjaminBeggs** [comment]
@casz I have implemented your suggestions. I think things are working properly now with the ConfiguredWithCode annotation. I am working on rewriting the unit test.

**Comment by BenjaminBeggs** [comment]
Export configuration unit tests have been fully rewritten @casz  @martinda

**Comment by BenjaminBeggs** [comment]
I have corrected some configuration-as-code.yaml indentations and removed the line for
`jnlp:`
from the commented out nodes configuration code (for https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68#discussion_r295124968). It was these things that were causing the test failures. It is unclear to me why the `jnlp:` line is causing a crash.

@martinda @casz

**Pull request merged**